felix/bpf-gpl/tc.c

```diff
@@ -263,6 +263,11 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 			ctx.state->flags |= CALI_ST_SKIP_FIB;
 		}
 		CALI_DEBUG("CT Hit\n");
+
+		if (ctx.state->ip_proto == IPPROTO_TCP && ct_result_is_syn(ctx.state->ct_result.rc)) {
+			CALI_DEBUG("Forcing policy on SYN\n");
+			goto syn_force_policy;
```

Comment:
This skips the socket cookie lookup; could that mean that policy is missing some information?

Reply:
:thinking: will have a look

Reply:
Yes, we should include that. We should only skip over NAT and RPF because that has been done as part of creating the CT entry that we just hit.